### DISCUSSION TITLE
MENU button does not work, second time page load on mobile devices

### DISCUSSION THREAD

**Discussion by mayelaatf** [uswds/uswds#4808] [Q&A]
Hi,
I hope you can provide some light to an issue I am having with the MENU button on mobile device.

Here is how I am using the USWDS :
Code base is in Drupal 9
Site has a primary and secondary navigation
Site uses uswds_base theme as base theme
Site is using USWDS 3.0.0

First time I navigate the site for the in a mobile device ( iphone / portrait orientation ),  I click the MENU button ( at the top right of the screen ) and a tab opens and shows the primary/secondary nav as expected. 
If I refresh the page and click again the MENU button nothing happen... and the footer secondary accordion menu tabs are messed up. ( all tabs are open ) 

It looks to me that there is a JS that gets unset when I reload the page. I do not know exactly how to debug this and I am wondering if you guys can provide some light on how to solve this issue.

I appreciate in advance any help on this,

Thanks,

Mayela

**Comment by mejiaj** [comment]
@mayelaatf can you provide a link to the site so we can take a look?

**Reply by mayelaatf** [reply]
@mejiaj , I can share you the behavior, via a zoom call , can you email me to coordinate ? - thanks in advance

**Reply by mejiaj** [reply]
Would you be able to provide a demo so we can see the problem? Are you getting any console errors in dev tools?

**Reply by mayelaatf** [reply]
@mejiaj , You can see the MENU issue here: https://www.youtube.com/watch?v=9Gp34pgxMl8

Also, a little bit of background here:
- When debugging this we see that the error points to this line of JS code:
`currentElement.nextElementSibling.setAttribute("id", menuId);`

- The above JS code is on the usa-footer component JS file :
   /usa-footer/src/index.js
   
- That code is not included in the uswds.min.js  but it is included in the uswds.js file

- why does not the uswds.min.js file include the above JS block of code?
- If we modify the above JS code to this:

`if (currentElement.nextElementSibling) {
  currentElement.nextElementSibling.setAttribute("id", menuId);}`

It fixes the issue.

**Reply by mejiaj** [reply]
Okay, thanks for providing this information. We'll look into it!

**Reply by mayelaatf** [reply]
An update here:
The previous workaround I posted before, does not work on real mobile devices. It worked only when using the chrome mobile simulator.
I found that by removing this:
t.nextElementSibling.setAttribute("id",e) from the uswds.min.js file fixes the issue, accordion tabs ( menu in the footer ) behave as expected and that the secondary menu button ( top right ) opens/closes as expected on mobile devices

**Reply by mejiaj** [reply]
@mayelaatf can you reproduce it in this component example - https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-header--default&viewMode=story?

I've checked in both browser in responsive mode and android device with Chrome:
- Website - https://designsystem.digital.gov/
- Header component preview - https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?id=components-header--default&args=&viewMode=story

**Reply by mayelaatf** [reply]
@mejiaj , I visited all the links you provided here. I tested the MENU button and it works as expected.
Please note :
1. My code has two menus :  
**Primary menu** : the one displays when clicking on the MENU btn (mobile) 
**Secondary menu** : Included in the footer and it is displayed as an accordion tabs in the footer (mobile) 
 2. I am using Drupal 9 / uswds_base theme / USWDS 3.0.0

**Reply by mejiaj** [reply]
@mayelaatf let's move the conversation to the issue so we can more accurately track.

**Comment by ryparker** [comment]
The missing follow-up is in [#4809](https://github.com/uswds/uswds/issues/4809#issuecomment-1488600852): the Drupal theme wrapped the footer heading in an extra element, so the heading no longer had the list as its next sibling.

The big footer expects this structure inside each navigation group:

```html
<h4 class="usa-footer__primary-link">Section</h4>
<ul class="usa-list usa-list--unstyled">
  <li><a href="/example">Example</a></li>
</ul>
```

I reproduced the reported null-sibling exception with USWDS 3.14.0 when that relationship is missing. With the heading immediately followed by its list, mobile initialization creates a working button whose `aria-controls` points to the list.

Check the rendered Drupal markup, including theme-added wrappers. Repairing that structure preserves the footer's accessible relationship. Removing the `setAttribute` line from the minified bundle suppresses the exception but also removes the ID used by `aria-controls`.

The minified code does contain the same operation, with shorter variable names, as your later example showed.
